packages/grafana-runtime/src/services/contextSrv.ts

```diff
@@ -0,0 +1,51 @@
+export interface User {
+  id: number;
+  isGrafanaAdmin: any;
+  isSignedIn: any;
+  orgRole: any;
+  orgId: number;
+  orgName: string;
+  login: string;
+  orgCount: number;
+  timezone: string;
+  helpFlags1: number;
+  lightTheme: boolean;
+  hasEditPermissionInFolders: boolean;
+  email?: string;
+}
+
+export interface ContextSrv {
+  pinned: any;
+  version: any;
+  user: User;
+  isSignedIn: any;
+  isGrafanaAdmin: any;
+  isEditor: any;
+  hasEditPermissionInFolders: boolean;
+  minRefreshInterval: string;
+  setLoggedOut: () => void;
+  hasRole: (role: string) => boolean;
+  isGrafanaVisible: () => boolean;
+  isAllowedInterval: (interval: string) => boolean;
+  getValidInterval: (interval: string) => string;
+  hasAccessToExplore: () => boolean;
+}
```

Comment:
Instead of exposing entire interface of the ContextSrv, please only expose public methods that I think are relevant to the consumers:

```
export interface ContextSrv {
  version: string;
  getUser(): User;
  hasRole(role:string): boolean;
  isGrafanaAdmin(): boolean;
  isEditor(): boolean;
}
```

The rest of the properties should be private to for internal usage.

Reply:
Wouldn't Typescript complain about that, since it doesn't match the instance class?

Reply:
of course it will :) But the class will only implement the interface, not the other way around. See for instance this working in the backenSrv: https://github.com/grafana/grafana/blob/1c49986b2f2e36179159258517696353a5ea01c8/public/app/core/services/backend_srv.ts#L31